### PR TITLE
Refactor Kryo serializer support to use chill/chill-java

### PR DESCRIPTION
**Pull request by ryanlecompte**
This pull request replaces Spark's currently custom `KryoSerializer` with one based on `chill` and `chill-java`. More details can be found here: https://github.com/mesos/spark/pull/590

I tested this with `java.util.Arrays.asList` and all seems to work well now that we have chill-java in place. Hopefully this is now in a state to be merged.

@rxin @mateiz @johnynek @sritchie

### CODE REVIEW

**Comment by mateiz** [comment]
Jenkins, add to whitelist

**Comment by mateiz** [comment]
Ryan, can you also update pom.xml with this?

**Comment by ryanlecompte** [comment]
Just updated `pom.xml` !

**Comment by rxin** [comment]
Is this what you are doing while you are on vacation? :)

File: core/src/main/scala/spark/KryoSerializer.scala [hunk 1]
```diff
@@ -19,24 +19,15 @@ package spark
...
-
+import com.twitter.chill.ScalaKryoInstantiator
 import serializer.{SerializerInstance, DeserializationStream, SerializationStream}
 import spark.broadcast._
 import spark.storage._
 
-private[spark]
-class KryoSerializationStream(kryo: Kryo, outStream: OutputStream) extends SerializationStream {
-
+private[spark] class KryoSerializationStream(kryo: Kryo, outStream: OutputStream) extends SerializationStream {
```
Comment (rxin) [inline core/src/main/scala/spark/KryoSerializer.scala]: wrap this so it is within 100 chars wide? same for KryoDeserializationStream

Comment (ryanlecompte) [inline core/src/main/scala/spark/KryoSerializer.scala]: fixed the wrapping! sorry about that, intellij auto-formats it for me.

**Comment by ryanlecompte** [comment]
hahaha, yes I really suck at vacations. :)

**Comment by sritchie** [comment]
Huh, funky that the style change caused this to fail. Transient test issue?

**Comment by rxin** [comment]
I tested this on Shark. Everything works: https://github.com/amplab/shark/compare/master...kryo

**Comment by ryanlecompte** [comment]
Nice!

**Comment by mateiz** [comment]
Cool, going to merge it then. I think that's a transient problem with that test. Thanks for putting this together.

**Pull request merged**